string[] labels
int8[] plot_ordering 
float32[] probabilities